File: DashboardBrasileirãoOO/DasboardBrasileirãoOOpython/src/modelo/time.py
class Time:
    def __init__(self, nome: str):
        self._nome = nome
        self._pontos = 0
        self._jogos = 0
        self._vitorias = 0
        self._empates = 0
        self._derrotas = 0
        self._gols_pro = 0
        self._gols_contra = 0
        self._saldo_gols = 0

    @property
    def nome(self):
        return self._nome

    @property
    def pontos(self):
        return self._pontos

    @property
    def jogos(self):
        return self._jogos

    @property
    def vitorias(self):
        return self._vitorias

    @property
    def empates(self):
        return self._empates

    @property
    def derrotas(self):
        return self._derrotas

    @property
    def gols_pro(self):
        return self._gols_pro

    @property
    def gols_contra(self):
        return self._gols_contra

    @property
    def saldo_gols(self):
        return self._saldo_gols

    def adicionar_vitoria(self, gols_pro: int, gols_contra: int):
        self._jogos += 1
        self._vitorias += 1
        self._pontos += 3
        self._gols_pro += gols_pro
        self._gols_contra += gols_contra
        self._atualizar_saldo_gols()

    def adicionar_empate(self, gols: int):
        self._jogos += 1
        self._empates += 1
        self._pontos += 1
        self._gols_pro += gols
        self._gols_contra += gols

    def adicionar_derrota(self, gols_pro: int, gols_contra: int):
        self._jogos += 1
        self._derrotas += 1
        self._gols_pro += gols_pro
        self._gols_contra += gols_contra
        self._atualizar_saldo_gols()

    def _atualizar_saldo_gols(self):
        self._saldo_gols = self.gols_pro - self.gols_contra

File: java/src/main/java/com/brasileirao/dashboard/modelo/Time.java

package com.brasileirao.dashboard.modelo;

import javafx.beans.property.SimpleIntegerProperty;
import javafx.beans.property.SimpleStringProperty;

public class Time {
    private final SimpleStringProperty nome;
    private final SimpleIntegerProperty pontos;
    private final SimpleIntegerProperty jogos;
    private final SimpleIntegerProperty vitorias;
    private final SimpleIntegerProperty empates;
    private final SimpleIntegerProperty derrotas;
    private final SimpleIntegerProperty golsPro;
    private final SimpleIntegerProperty golsContra;
    private final SimpleIntegerProperty saldoGols;

    public Time(String nome) {
        this.nome = new SimpleStringProperty(nome);
        this.pontos = new SimpleIntegerProperty(0);
        this.jogos = new SimpleIntegerProperty(0);
        this.vitorias = new SimpleIntegerProperty(0);
        this.empates = new SimpleIntegerProperty(0);
        this.derrotas = new SimpleIntegerProperty(0);
        this.golsPro = new SimpleIntegerProperty(0);
        this.golsContra = new SimpleIntegerProperty(0);
        this.saldoGols = new SimpleIntegerProperty(0);
    }

    // Getters para os valores 
    public String getNome() { return nome.get(); }
    public int getPontos() { return pontos.get(); }
    public int getJogos() { return jogos.get(); }
    public int getVitorias() { return vitorias.get(); }
    public int getEmpates() { return empates.get(); }
    public int getDerrotas() { return derrotas.get(); }
    public int getGolsMarc() { return golsPro.get(); }
    public int getGolsSofr() { return golsContra.get(); }
    public int getSaldoGols() { return saldoGols.get(); }

    // MÉTODOS DE PROPRIEDADE 
    // A TableView usa estes métodos para se conectar aos dados
    public SimpleStringProperty nomeProperty() { return nome; }
    public SimpleIntegerProperty pontosProperty() { return pontos; }
    public SimpleIntegerProperty jogosProperty() { return jogos; }
    public SimpleIntegerProperty vitoriasProperty() { return vitorias; }
    public SimpleIntegerProperty empatesProperty() { return empates; }
    public SimpleIntegerProperty derrotasProperty() { return derrotas; }
    public SimpleIntegerProperty golsProProperty() { return golsPro; }
    public SimpleIntegerProperty golsContraProperty() { return golsContra; }
    public SimpleIntegerProperty saldoGolsProperty() { return saldoGols; }

    // Métodos para atualizar as estatísticas
    public void adicionarVitoria(int golsPro, int golsContra) {
        this.jogos.set(getJogos() + 1);
        this.vitorias.set(getVitorias() + 1);
        this.pontos.set(getPontos() + 3);
        this.golsPro.set(getGolsMarc() + golsPro);
        this.golsContra.set(getGolsSofr() + golsContra);
        atualizarSaldoGols();
    }

    public void adicionarEmpate(int gols) {
        this.jogos.set(getJogos() + 1);
        this.empates.set(getEmpates() + 1);
        this.pontos.set(getPontos() + 1);
        this.golsPro.set(getGolsMarc() + gols);
        this.golsContra.set(getGolsSofr() + gols);
    }

    public void adicionarDerrota(int golsPro, int golsContra) {
        this.jogos.set(getJogos() + 1);
        this.derrotas.set(getDerrotas() + 1);
        this.golsPro.set(getGolsMarc() + golsPro);
        this.golsContra.set(getGolsSofr() + golsContra);
        atualizarSaldoGols();
    }

    private void atualizarSaldoGols() {
        this.saldoGols.set(getGolsMarc() - getGolsSofr());
    }
}